zuul-core/src/main/java/com/netflix/zuul/message/ZuulMessageImpl.java

```diff
@@ -147,16 +147,13 @@ public byte[] getBody() {
             return null;
         }
 
-        int size = 0;
-        for (final HttpContent chunk : bodyChunks) {
-            size += chunk.content().readableBytes();
-        }
+        int size = this.getBodyLength();
         final byte[] body = new byte[size];
         int offset = 0;
         for (final HttpContent chunk : bodyChunks) {
             final ByteBuf content = chunk.content();
-            final int len = content.readableBytes();
-            content.getBytes(content.readerIndex(), body, offset, len);
+            final int len = content.writerIndex(); // writer idx tracks the total readable bytes in the buffer
```

Comment:
If we do this, let's at least indicate in the comments that the API shall always return the complete buffered body, and not the pending readable bytes. Same for `getBodyLength()`